File: plugins/setup/src/main/java/net/elytrarace/setup/conversation/cup/CupDisplayNamePrompt.java
package net.elytrarace.setup.conversation.cup;

import net.elytrarace.api.conversation.ConversationContext;
import net.elytrarace.api.conversation.Prompt;
import net.elytrarace.api.conversation.StringPrompt;
import net.kyori.adventure.text.Component;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import java.util.Optional;

public class CupDisplayNamePrompt extends StringPrompt {
    @Override
    public @NotNull Component getPromptText(@NotNull ConversationContext context) {
        String name = Optional.ofNullable((String) context.getSessionData("cupName")).orElse("No name");
        return Component.translatable("prompt.cup.displayname").arguments(Component.text(name));
    }

    @Override
    public @Nullable Prompt acceptInput(@NotNull ConversationContext context, @Nullable String input) {
        if (input == null || input.isEmpty()) {
            return this;
        }
        context.setSessionData("displayname", input);
        return new CupSetupFinish();
    }
}

File: src/main/kotlin/net/elytrarace/conversation/cup/CupDisplayNamePrompt.kt
package net.elytrarace.conversation.cup

import net.elytrarace.conversation.ConversationContext
import net.elytrarace.conversation.FinishPrompt
import net.elytrarace.conversation.Prompt
import net.elytrarace.conversation.StringPrompt
import net.elytrarace.model.dbo.Cup
import net.kyori.adventure.text.Component
import net.kyori.adventure.text.minimessage.MiniMessage
import org.jetbrains.exposed.sql.transactions.transaction

class CupDisplayNamePrompt : StringPrompt() {
    override fun getPromptText(context: ConversationContext): Component {
        val name = context.getSessionData("cup_name") as String
        return Component.translatable("prompt.cup.name.display").args(Component.text(name))
    }

    override fun acceptInput(context: ConversationContext, input: String?): Prompt? {
        context.setSessionData("cup_display_name", input)
        transaction {
            Cup.new {
                name = context.getSessionData("cup_name") as String
                displayName = context.getSessionData("cup_display_name") as String
            }
        }
        return FinishPrompt()
    }
}